src/extensions/insights/all-insights/duplicate-dependencies.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { Color, Box, Text, render } from 'ink';
+import { Insight, InsightResult, RawResult } from '../insight';
+import { ComponentGraph } from '../../graph/component-graph';
+import { VersionSubgraph } from '../../graph/duplicate-dependency';
+
+export const INSIGHT_NAME = 'duplicateDependencies';
+
+type FormattedEntry = {
+  dependencyId: string,
+  latestVersion: string,
+  dependents: {
+    id: string,
+    usedVersion: string
+  }[]
+}
+export default class DuplicateDependencies implements Insight {
+  name = INSIGHT_NAME;
+  description = 'Get all duplicate dependencies in component graph';
+  graph: ComponentGraph;
+  constructor(graph: ComponentGraph) {
+    this.graph = graph;
+  }
+  async _runInsight(): Promise<RawResult> {
+    const duplicates = this.graph.findDuplicateDependencies();
+    return {
+      message: `Found ${[...duplicates.keys()].length} duplicate dependencies.`,
+      data: duplicates
+    };
+  }
+
+  _formatData(data: any): FormattedEntry[] {
+    let formatted: FormattedEntry[] = [];
+    for (const [dependency, depData] of data.entries()){
+      const dependents: {id: string, usedVersion: string}[] = this._getDependents(depData.priorVersions);
+      formatted.push({
+        dependencyId: dependency,
+        latestVersion: depData.latestVersionId,
+        dependents: dependents
+      })
+    }
+    return formatted;
+  }
+
+  _getDependents(priorVersions: VersionSubgraph[]): {id: string, usedVersion: string}[]{
```

Comment:
Make the `{id: string, usedVersion: string}` a declared type (you are using it twice)

Reply:
fixed